### PR TITLE
refactor(charts): move registry secret from renku repo

### LINKED ISSUE #167: make the registry secret in the notebooks chart
The secret used for pulling private images is currently created in the main renku chart - we should move it to the notebooks service chart instead.

### PR DESCRIPTION
**Pull request by rokroskar**
Note that this change allows non-root admin tokens to be used. This should make it easier to rotate them periodically. 

```
gitlab:
  registry:
    username: a_user
    token: 12345
```

closes #167

### CODE REVIEW

File: helm-chart/renku-notebooks/templates/registry-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{{- $auth := printf "%s:%s" .Values.gitlab.registry.username .Values.gitlab.registry.token | b64enc -}}
+{{- $registry := .Values.gitlab.registry.host | replace "http://" "" | replace "https://" "" | replace ":443" "" -}}
+{{- $secret := printf "{\"auths\":{\"%s\":{\"username\":\"%s\",\"password\":\"%s\",\"email\":\"root@example.com\",\"auth\":\"%s\"}}}" $registry .Values.gitlab.registry.username .Values.gitlab.registry.token $auth -}}
```
Comment (ableuler) [inline helm-chart/renku-notebooks/templates/registry-secret.yaml]: Does that email have to be here? It could be any admin, no?

File: helm-chart/renku-notebooks/values.yaml [hunk 2]
```diff
@@ -40,26 +42,30 @@ securityContext:
 # turns on flask debug mode
 debug: true
 
+## Default server options - these will be provided to the user in the UI
```
Comment (ableuler) [inline helm-chart/renku-notebooks/values.yaml]: I think this should be in a separate PR.

File: helm-chart/renku-notebooks/values.yaml [hunk 3]
```diff
@@ -178,16 +177,30 @@ jupyterhub:
...
       ## use openssl rand -hex 32 to generate clientSecret
       clientSecret:
-      callbackUrl: '""' # Forces the default callback url
+      callbackUrl: # Leaving it null forces the default callback url
   proxy:
     service:
       type: ClusterIP
     https:
       enabled: false
+  ## Enable idle user server culling
```
Comment (ableuler) [inline helm-chart/renku-notebooks/values.yaml]: Also material for a separate PR.

File: helm-chart/renku-notebooks/templates/registry-secret.yaml (hunk 1 above)
Comment (rokroskar) [reply helm-chart/renku-notebooks/templates/registry-secret.yaml]: true I hadn't noticed that. It doesn't actually seem to match the email in gitlab for the root user so it doesn't look like it's being used. I would leave it for now.

File: helm-chart/renku-notebooks/values.yaml (hunk 2 above)
Comment (rokroskar) [reply helm-chart/renku-notebooks/values.yaml]: hmm... just adding a few lines of documentation to the values file... but ok.

Comment (ableuler) [reply helm-chart/renku-notebooks/values.yaml]: Sorry, I thought that had ended up in here by accident because it has nothing to do with the title/description of the PR or the commit message. I'm perfectly fine with having "non-atomic" PRs, but at least having two commits for things that have nothing in common would indicate that this was intended.

Comment (rokroskar) [reply helm-chart/renku-notebooks/values.yaml]: no worries, I already have a new PR for that, can include more documentation (and fix the README) there as well

**Comment by rokroskar** [comment]
I tested this on my dev environment using my own token and launching a project from one of my private repos - seems to work as intended.
